contracts/UnipoolTokenDistributor.sol

```diff
@@ -173,13 +173,23 @@ contract UnipoolTokenDistributor is LPTokenWrapper, OwnableUpgradeable {
 
     function notifyRewardAmount(uint256 reward) external onlyRewardDistribution updateReward(address(0)) {
         uint256 _timestamp = getTimestamp();
+        uint256 _reward;
         if (_timestamp >= periodFinish) {
-            rewardRate = reward.div(duration);
+            _reward = reward;
         } else {
             uint256 remaining = periodFinish.sub(_timestamp);
             uint256 leftover = remaining.mul(rewardRate);
-            rewardRate = reward.add(leftover).div(duration);
+            _reward = reward.add(leftover);
         }
+
+        (bool success, bytes memory data) =
```

Comment:
So this function is now adding an additional check if the unipool has enough balance that is equal or greater than the amount of rewards it will distribute? 

the internal variable _reward does it decrease as rewards are distributed or is it a total number of rewards it has given across its entire lifespan?